### PR TITLE
Open-source extractMetaNetDef & runGlobalInitialization, add new Predictor constructor from db file, and add run_map_outputs

### PR DESCRIPTION
**Pull request by theweiho**
1. Open-source extractMetaNetDef and runGlobalInitialization, for use in
2. new Predictor constructor from db file.
3. Add new run function that returns outputs as TensorMap.

### CODE REVIEW

**Review by salexspb — APPROVED** [review]
seems legit. Looking forward our next steps on improving these abstractions. 
To give context to others - we had a long discussion about this, basically we need to provide DB support in OSS asap for a release. Once done (in a few days), plan is to clean up the db support and provide unified predictor abstraction. Then predictors could be interchangeable (threaded and non threaded, memory managing or not, etc).

File: caffe2/core/predictor.cc [hunk 1]
```diff
@@ -49,13 +49,16 @@ const NetDef& getNet(const MetaNetDef& def, const std::string& name) {
...
-    const std::string& name) {
+    const std::string& name,
+    const bool optional = false) {
   for (const auto& b : def.blobs()) {
     if (b.key() == name) {
       return b.value();
     }
   }
-  CAFFE_THROW("Blob not found: ", name);
+  CAFFE_ENFORCE(optional, "Blob not found: ", name);
```
Comment (hlu1) [inline caffe2/core/predictor.cc]: Why do you need to make this `CAFFE_ENFORCE` optional?

Comment (salexspb) [reply caffe2/core/predictor.cc]: hm, agree with @hlu1 , this seems weird. Could you try getting rid of this, please?

Comment (theweiho) [reply caffe2/core/predictor.cc]: @hlu1 confirmed that real models should always have the outputs blob, so updated predictor_test.cc instead.

**Comment by theweiho** [comment]
@hlu1, @salexspb - it seems enforcing that models have an outputs blob causes some build failures? I had originally thought those were not related to my change originally, but they passed when I made the outputs optional... Or just unlucky flakiness?

**Comment by theweiho** [comment]
Apologies to the newly added reviewers - seems like it was triggered by an accidental merge that has now been reverted. The UI doesn't seem to allow me to remove reviewers, so please ignore

**Pull request merged**